apps/framework-cli/src/framework/transform.ts

```diff
@@ -26,9 +44,28 @@ const scrubVersionFromTopic = (topic: string): string => {
 };
 const version = getVersion();
 
+const getSaslConfig = (): SASLOptions | undefined => {
+  const mechanism = SASL_MECHANISM ? SASL_MECHANISM.toLowerCase() : "";
+  switch (mechanism) {
+    case "plain":
+    case "scram-sha-256":
+    case "scram-sha-512":
+      return {
+        mechanism: mechanism,
+        username: SASL_USERNAME || "",
+        password: SASL_PASSWORD || "",
+      };
+    default:
+      console.log(`Unsupported SASL mechanism: ${SASL_MECHANISM}`);
+      return undefined;
+  }
+};
+
 const kafka = new Kafka({
   clientId: "deno-consumer",
-  brokers: ["redpanda:9092"],
+  brokers: [BROKER],
+  ssl: SECURITY_PROTOCOL === "SASL_SSL",
+  sasl: getSaslConfig(),
```

Comment:
Tested this locally without sasl. Also remotely by connecting to our internal Redpanda cluster.